2016-10-29  Avery Wong  <classave@lnxsrv09.seas.ucla.edu>

        * timespec.h (timespec_cmp):fixed timespec_cmp by using if else
        loops instead of subtraction to avoid overflow issues

--- coreutils-with-bug/lib/timespec.h	2005-09-21 23:05:39.000000000 -0700
+++ coreutils-with-bug-patched/lib/timespec.h	2016-10-27 22:50:19.851220000 -0700
@@ -46,7 +46,21 @@
 timespec_cmp (struct timespec a, struct timespec b)
 {
   int diff = a.tv_sec - b.tv_sec;
-  return diff ? diff : a.tv_nsec - b.tv_nsec;
+  // return diff ? diff : a.tv_nsec - b.tv_nsec;
+ 
+if (diff > 0)
+{
+return 1;
+}
+else if(diff < 0)
+{
+  return -1;
+}
+else
+{
+return 0;
+}
+ 
 }
 
 # if ! HAVE_DECL_NANOSLEEP
